File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        char[] sc = s.toCharArray();
        int n = sc.length;
        int i=0;
        for(char c:t.toCharArray()) {
            if(i==n)
                return true;
            if(sc[i] == c) {
                i++;
                
            }
        }
        return i == n;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n = s.length();
        int i=0;
        for(int j=0;j<t.length();j++) {
            if(i==n)
                return true;
            if(s[i] == t[j]) {
                i++;
            }
        }
        return i == n;
    }
};